nixos/modules/services/monitoring/snmpd.nix

```diff
@@ -0,0 +1,60 @@
+{ pkgs, config, lib, ... }:
+
+let
+  cfg = config.services.snmpd;
+in {
+  options.services.snmpd = {
+    enable = lib.mkEnableOption "snmpd";
+
+    package = lib.mkPackageOption pkgs "net-snmp" {};
+
+    listenAddress = lib.mkOption {
+      type = lib.types.str;
+      default = "0.0.0.0";
+      description = lib.mdDoc ''
+        The address to listen on for SNMP and AgentX messages
+      '';
+      example = "127.0.0.1";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 161;
+      description = lib.mdDoc ''
+        The port to listen on for SNMP and AgentX messages
+      '';
+    };
+
+    openFirewall = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Open port in firewall for snmpd
+      '';
+    };
+
+    configFile = lib.mkOption {
+      type = lib.types.path;
+      description = lib.mdDoc ''
+        Path to the snmpd.conf file.
+      '';
+    };
```

Comment:
Must this be a file? Woudln't it be easier to write the config in string form?

Reply:
Can the secrets not being loaded via envs or so? I mean we could work around this but just throwing the entire file into sops or so is indeed easier.

Reply:
@SuperSandro2000 , I could try adding environment variables support if you think it would be a good idea, however I would appreciate an example service which implements this.

In the meantime, I've just pushed another change, to add myself as the service maintainer.